stdlib/public/Cxx/CxxConvertibleToCollection.swift

```diff
@@ -21,36 +21,36 @@ public protocol CxxConvertibleToCollection {
   mutating func __endUnsafe() -> RawIterator
 }
 
-@inlinable @inline(__always)
-internal func forEachElement<C: CxxConvertibleToCollection>(
-  of c: C,
-  body: (C.RawIterator.Pointee) -> Void
-) {
-  var mutableC = c
-  withExtendedLifetime(mutableC) {
-    var rawIterator = mutableC.__beginUnsafe()
-    let endIterator = mutableC.__endUnsafe()
-    while rawIterator != endIterator {
-      body(rawIterator.pointee)
-      rawIterator = rawIterator.successor()
+extension CxxConvertibleToCollection {
+  @inlinable
+  internal func forEach(_ body: (RawIterator.Pointee) -> Void) {
+    var mutableSelf = self
+    withExtendedLifetime(mutableSelf) {
+      var rawIterator = mutableSelf.__beginUnsafe()
+      let endIterator = mutableSelf.__endUnsafe()
+      while rawIterator != endIterator {
+        body(rawIterator.pointee)
+        rawIterator = rawIterator.successor()
+      }
     }
   }
 }
 
-extension Array {
-  /// Creates an array containing the elements of a C++ container.
+extension RangeReplaceableCollection {
+  /// Creates a collection containing the elements of a C++ container.
   ///
-  /// This initializes the array by copying every element of the C++ container.
+  /// This initializes the collection by copying every element of the C++
+  /// container.
   ///
   /// - Complexity: O(*n*), where *n* is the number of elements in the C++
   ///   container when each element is copied in O(1). Note that this might not
   ///   be true for certain C++ types, e.g. those with a custom copy
   ///   constructor that performs additional logic.
-  public init<C: CxxConvertibleToCollection>(_ c: C)
+  public init<C: CxxConvertibleToCollection>(contentsOf c: C)
     where C.RawIterator.Pointee == Element {
```

Comment:
I think we should make `Pointee` the primary associated type of `CxxConvertibleToCollection`.

It would need to promoted to an associated type, but I think it would very much be worth doing that, if only to allow us to use the far friendlier new-style declaration syntax in this and similar common cases:

```swift
public init(contentsOf c: some CxxConvertibleToCollection<Element>) {
  ...
}
```

This would require the following change to the protocol definition:

```swift
protocol CxxConvertibleToCollection<Pointee> {
  associatedtype Pointee
  associatedtype RawIterator: UnsafeCxxInputIterator where RawIterator.Pointee == Pointee

  mutating func __beginUnsafe() -> RawIterator
  mutating func __endUnsafe() -> RawIterator
}
```

(This would be very similar to how `protocol Sequence` works.)

Similarly, `protocol CxxSequence` should designate `Element` as its primary associated type:

```
public protocol CxxSequence<Element>: Sequence { ... }
```